src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,52 @@
+package duke;
+
+import java.io.FileNotFoundException;
+
+import duke.command.Command;
+import javafx.stage.Stage;
+
+/**
+ * Duke is a Personal Assistant Chatbot that helps a person to keep track of various things.
+ *
+ * @author Samsation
+ * @version CS2103T AY 22/23 Sem 1
+ *
+ */
+
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private boolean isTerminated = false;
+
+    /**
+     * A constructor for Duke.
+     *
+     * @param stage The current Stage of the Application.
+     */
+    public Duke(Stage stage) {
+        this.ui = new Ui(stage);
```

Comment:
I think you could consider leaving the `this` keyword out, according to the following coding standard rule:
> Use the this keyword only when a field is shadowed by a method or constructor parameter.